frontend/src/pages/Task.tsx

```diff
@@ -0,0 +1,624 @@
+import { Link, useNavigate, useParams } from "react-router-dom";
+import { useState } from "react";
+import {
+    useGetModelsQuery,
+    useGetTasksQuery,
+    useRenameTaskMutation,
+    useUpdateGrammarModeMutation,
+    useUpdateTaskInputsMutation,
+    useUpdateTaskModelMutation,
+    useUpdateTaskPromptMutation,
+} from "../api/services/v1";
+import { compile, serializeGrammar } from "@intrinsicai/gbnfgen";
+import ts from "typescript";
+
+import Page from "../components/layout/Page";
+import TwoColumnLayout from "../components/layout/TwoColumnLayout";
+import Column from "../components/layout/Column";
+import Card from "../components/core/Card";
+import Button from "../components/core/Button";
+import Callout from "../components/core/Callout";
+import { TaskInfo } from "../api";
+import Dropdown from "../components/core/Dropdown";
+import { Icon } from "@blueprintjs/core";
+
+function TaskHeader({ task }: { task: string }) {
+    const navigate = useNavigate();
+    const [isEditing, setEditing] = useState<boolean>(false);
+    const [taskName, setTaskName] = useState<string>(task);
```

Comment:
NB: the `<>` in the useState are optional here b/c the type can be inferred by the value you're passing in.

The only time you actually need to specify the type between `<>` is when the initial value is `undefined`